src/utils/buildElementDefinitions.js

```diff
@@ -1,17 +1,36 @@
-const buildElementDefinitions = (elements, constructor) => elements.reduce(
-  (aggregate, element) => {
-    if (element.defineId) {
-      // eslint-disable-next-line no-param-reassign
-      aggregate[element.defineId] = element;
-    } else {
-      element.promise.then((clazz) => {
-        if (constructor.registry.get(element.name) === undefined) {
-          constructor.registry.define(element.name, clazz);
+const buildElementDefinitions = (elements, constructor) => {
+  const promises = [];
+  const definitions = elements.reduce(
+    (aggregate, element) => {
+      if (typeof element === 'string') {
+        const clazz = customElements.get(element);
+        if (clazz) {
+          // eslint-disable-next-line no-param-reassign
+          aggregate[element] = clazz;
+        } else {
+          promises.push(customElements.whenDefined(element).then(() => {
+            if (constructor.registry.get(element) === undefined) {
+              constructor.registry.define(element, customElements.get(element));
+            }
+          }));
         }
+      } else {
+        // eslint-disable-next-line no-param-reassign
+        aggregate[element.defineId] = element;
+      }
+      return aggregate;
+    }, {},
+  );
+  // eslint-disable-next-line no-param-reassign
+  constructor.elementDefinitionsLoaded = promises.length === 0;
```

Comment:
Looks hackish and not very optimal, but working

Reply:
What is not optimal about it?

Reply:
I think you can use a counter instead of `Promise.all`. And the `constructor.elementDefinitionsLoaded` is not a good way. It is better to extend some class that implements the `elementDefinitionsLoaded` getter.

Reply:
a counter will add more logic as I want to keep the elementDefinitionsLoaded a boolean and not a function for simplicity. About extending a class, this getter is static and there is no inheritance for static members

Reply:
> a counter will add more logic
```
counter++;
customElements.whenDefined(element).then(() => {
    counter--;
    if (constructor.registry.get(element) === undefined) {
        constructor.registry.define(element, customElements.get(element));
    }
    constructor.elementDefinitionsLoaded = counter === 0;
});
```

> this getter is static and there is no inheritance for static members
You can use not static `this.elementDefinitionsLoaded` in render, it's not problem

Reply:
I already released and I don't think it will make that much difference, you can update now to see if it works in the official released verison